dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/SubscriptionRestRepository.java

```diff
@@ -0,0 +1,292 @@
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.Subscription;
+import org.dspace.eperson.SubscriptionParameter;
+import org.dspace.eperson.service.EPersonService;
+import org.dspace.eperson.service.SubscribeService;
+import org.dspace.subscriptions.SubscriptionEmailNotificationService;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.rest.webmvc.ResourceNotFoundException;
+import org.springframework.hateoas.Link;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Component;
+
+/**
+ * This is the repository responsible to manage SubscriptionRest object
+ *
+ * @author Mykhaylo Boychuk (mykhaylo.boychuk at 4science.com)
+ */
+@Component(SubscriptionRest.CATEGORY + "." + SubscriptionRest.NAME)
+public class SubscriptionRestRepository extends DSpaceRestRepository<SubscriptionRest, Integer>
+                                         implements InitializingBean {
+
+    @Autowired
+    private ConverterService converter;
+    @Autowired
+    private EPersonService ePersonService;
+    @Autowired
+    private AuthorizeService authorizeService;
+    @Autowired
+    private SubscribeService subscribeService;
+    @Autowired
+    private DSpaceObjectUtils dspaceObjectUtil;
+    @Autowired
+    private DiscoverableEndpointsService discoverableEndpointsService;
+    @Autowired
+    private SubscriptionEmailNotificationService subscriptionEmailNotificationService;
+
+    @Override
+    @PreAuthorize("hasPermission(#id, 'subscription', 'READ')")
+    public SubscriptionRest findOne(Context context, Integer id) {
+        Subscription subscription = null;
+        try {
+            subscription = subscribeService.findById(context, id);
+        } catch (SQLException e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+        return Objects.isNull(subscription) ? null : converter.toRest(subscription, utils.obtainProjection());
+    }
+
+    @Override
+    @PreAuthorize("hasAuthority('ADMIN')")
+    public Page<SubscriptionRest> findAll(Context context, Pageable pageable) {
+        try {
+            List<Subscription> subscriptionList = subscribeService.findAll(context, null,
+                                                           Math.toIntExact(pageable.getPageSize()),
+                                                           Math.toIntExact(pageable.getOffset()));
+            Long total = subscribeService.countAll(context);
+            return converter.toRestPage(subscriptionList, pageable, total,  utils.obtainProjection());
+        } catch (Exception e) {
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
+    @SearchRestMethod(name = "findByEPerson")
+    @PreAuthorize("hasPermission(#epersonId, 'AdminOrOwner', 'READ')")
```

Comment:
you can simply rely on the READ permission on the eperson, no need to duplicate the logic in a dedicated plugin